addons/mail/tests/common.py

```diff
@@ -610,6 +640,22 @@ def assertSentEmail(self, author, recipients, **values):
 
         return sent_mail
 
+    # ------------------------------------------------------------
+    # TOOLS
+    # ------------------------------------------------------------
+
+    def get_message_count_per_record(self, records):
+        domain = expression.OR([
+            [('model', '=', record._name), ('res_id', '=', record.id)]
+            for record in records
+        ])
```

Comment:
We only ever use recordsets with this one, why not make the assumption we get a recordset and just make the domain `records.ids`?

Either way, even for psql I think it's more efficient to call this multiple times if you have different types of records for some reason, rather than having a large OR block